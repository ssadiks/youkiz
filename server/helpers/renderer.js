import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import serialize from 'serialize-javascript';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Routes from '../../src/Routes';
import HeaderContainer from '../../src/components/container/HeaderContainer/HeaderContainer';

const renderer = (req, store) => {
  const muiTheme = getMuiTheme({
    userAgent: req.headers['user-agent']
  });

  const content = renderToString(
    <Provider store={store}>
      <MuiThemeProvider muiTheme={muiTheme}>
        <StaticRouter location={req.path} context={{}}>
          <div>
            {renderRoutes(Routes)}
          </div>
        </StaticRouter>
      </MuiThemeProvider>
    </Provider>
  );

  return `
    <html>
      <head>
        <link href="css/style.css" rel="stylesheet">
      </head>
      <body>
        <div id="root">${content}</div>
        <script>
          window.INITIAL_STATE = ${serialize(store.getState())}
        </script>
        <script src="js/vendors.js"></script>
        <script src="js/bundle.js"></script>
      </body>
    </html>
  `;
};

export default renderer;
