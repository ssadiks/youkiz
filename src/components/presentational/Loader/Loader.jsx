import React from 'react';
import RefreshIndicator from 'material-ui/RefreshIndicator';

const Loader = () => {
  const style = {
    refresh: {
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)'
    }
  };

  return (
    <div className="Loader">
      <RefreshIndicator
        size={40}
        left={10}
        top={0}
        status="loading"
        style={style.refresh}
        className="Loader__refresh"
      />
    </div>
  );
};

export default Loader;
