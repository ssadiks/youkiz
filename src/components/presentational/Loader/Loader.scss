.Loader {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.7);

  &.lazy-loading {
    z-index: 9;
    bottom: 5rem;
    top: inherit;
    left: 50%;
    width: inherit;
    height: inherit;
  }

  &__refresh {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
}
