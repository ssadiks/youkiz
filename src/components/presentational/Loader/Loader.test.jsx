import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './Loader';

jest.mock('material-ui/RefreshIndicator', () => 'RefreshIndicator');

describe('Loader render', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Loader />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
