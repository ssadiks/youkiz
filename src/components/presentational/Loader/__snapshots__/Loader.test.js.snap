// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader render renders correctly 1`] = `
<div
  className="Loader"
>
  <RefreshIndicator
    className="Loader__refresh"
    left={10}
    size={40}
    status="loading"
    style={
      Object {
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      }
    }
    top={0}
  />
</div>
`;
